native/screens/SignIn.jsx

```diff
@@ -1,103 +1,166 @@
-// import { useEffect, useState } from "react";
-// import { useNavigate } from "react-router-dom";
-// // assets
-// import { useDispatch, useSelector } from "react-redux";
-// import { loginUser } from "../../services/redux/slices/uiActions.js";
-// import "../SignUp/SignUp.scss";
-// import { handleChange } from "../../services/utils/formHandlers";
-// import { SingleActionButton } from "../components/Button/SingleActionButton";
-// import { checkEmailAuth, signOut, verify } from "../../services/api/users";
-import { StyleSheet, Text, View } from "react-native";
+import { useState, createRef, useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { signOut, checkEmailAuth, verify } from '../services/api/users.js';
+import { loginUser } from '../services/redux/actions/uiActions.js';
+
+// Native Components
+import { 
+  StyleSheet, 
+  Text, 
+  View, 
+  TextInput, 
+  TouchableOpacity, 
+  Button,
+  Alert,
+} from "react-native";
 
-export const SignIn = () => {
-  return (
-    <View>
-      <Text>SignIn</Text>
-    </View>
-  );
-};
 
-// export const ReactSignIn = () => {
-//   const navigate = useNavigate();
-//   const dispatch = useDispatch();
-//   const { _id: userId } = useSelector((state) => state.ui.user);
-//   const [authError, setAuthError] = useState(null);
-//   const [noAccountError, setNoAccountError] = useState(null);
-//   const [loginInfo, setLoginInfo] = useState({
-//     email: "test@test.com",
-//     password: "test",
-//   });
+export const SignIn = ({navigation}) => {
+  // redux
+  const dispatch = useDispatch();
+  // error states
+  const [noAccountError, setNoAccountError] = useState(null);
+  const [authError, setAuthError] = useState(null);
+  // log in state
+  const [logInInfo, setLogInInfo] = useState({
+    email: '',
+    password: '',
+  });
+  // input references
+  const { emailInputRef, passwordInputRef } = createRef();
 
-//   useEffect(() => {
-//     signOut();
-//   }, []);
+  // sign out user on sign in screen render
+  useEffect(() => {
+    signOut();
+  }, []);
 
-//   const handleSignIn = async (event) => {
-//     event.preventDefault();
-//     await dispatch(loginUser(loginInfo));
-//     const user = await verify();
-//     if (user.email === loginInfo.email) {
-//       navigate(`/users/${userId}/edit`);
-//     } else {
-//       setAuthError(
-//         "Invalid credentials. Please check your details and try again."
-//       );
-//       setLoginInfo((prevState) => {
-//         return {
-//           ...prevState,
-//           password: "",
-//         };
-//       });
-//     }
-//   };
+  const handleSignIn = async () => {
+    await dispatch(loginUser(logInInfo));
+    const user = await verify();
```

Comment:
curious about this because we are dispatching the login info before we verify that their information is correct
if it works, it works, i'm curious about if there may be a better solution. ping me if you'd like to put our heads together on it

Reply:
When I try to use the user._id from useSelector to reroute to EditProfile, the state is not updated after dispatching loginUser.

I've been using verify here to get the user id via the token since that proves to be updated appropriately during login and returns the most recent signed in user.

I've done some investigating into this and have had trouble getting a clear understanding of a solution, but maybe your understanding of redux would help clean some things up.

Reply:
I just finished looking into this. Since React waits for the function execution to finish before updating the redux state, we won't be able to call the API, update the state, then check it in the same function. None of that information would be available until the next render.

Therefore, i deprecated that `loginUser` function in redux and used the actual `signIn`  function from `services/api` so that we can actually verify the response from the api before we update the redux store. It was a weird React-state lifecycle thing but good initial solution!